sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/graph/PipelineExecution.scala

```diff
@@ -46,6 +46,10 @@ class PipelineExecution(context: PipelineUpdateContext) {
   def startPipeline(): Unit = synchronized {
     // Initialize the graph.
     val resolvedGraph = resolveGraph()
+    if (context.fullRefreshTables.nonEmpty) {
+      State.reset(resolvedGraph, context)
```

Comment:
does this need to happen after `DatasetManager.materializeDatasets`, seems like we only know `normalizedPath` after `materializeTable` [here](https://github.com/sryza/spark/blob/ac211435c7a4ca1a34a52e61c0647355ebd29476/sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/graph/DatasetManager.scala#L210)